internal/pkg/test/tool/require.go

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2019, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package tool
+
+import (
+	"os/exec"
+	"sync"
+	"syscall"
+	"testing"
+
+	"github.com/sylabs/singularity/pkg/util/fs/proc"
+)
+
+var hasUserNamespace bool
+var hasUserNamespaceOnce sync.Once
+
+// RequireUserNamespace checks that the current test could
+// use user namespace, if user namespaces are not enabled or
+// supported, the current test is skipped with a message.
+func RequireUserNamespace(t *testing.T) {
+	// not performance critical, just save extra execution
+	// to get the same result
+	hasUserNamespaceOnce.Do(func() {
+		// user namespace is a bit special, as there is no simple
+		// way to detect if it's supported or enabled via a call
+		// on /proc/self/ns/user, the easiest and reliable way seems
+		// to directly execute a command by requesting user namespace
+		cmd := exec.Command("/bin/true")
+		cmd.SysProcAttr = &syscall.SysProcAttr{
+			Cloneflags: syscall.CLONE_NEWUSER,
+		}
+		err := cmd.Run()
+		if err != nil {
+			t.Logf("error while checking user namespace support: %s", err)
+		}
+		// no error means user namespaces are enabled
+		hasUserNamespace = err == nil
+	})
+	if !hasUserNamespace {
+		t.Skipf("user namespaces seems not enabled or supported")
+	}
+}
+
+// RequireFilesystem checks that the current test could
+// use the corresponding filesystem, if the filesystem is
+// not listed in /proc/filesystems, the current test is
+// skipped with a message.
+func RequireFilesystem(t *testing.T, fs string) {
```

Comment:
this is fine, but I'm not sure I understand the need for it. I can see that you are using this below to check for cgroup, but I suspect this is going beyond that.

Also, from this PR alone, this is a helper for the next function. I would not export it. I would instead make wrappers like the below one. But then again, I don't know the actual use case for this...

Reply:
useful for e2e tests using overlay: `RequireFilesytem(t, "overlay")`